pkg/applier/applier_test.go

```diff
@@ -391,6 +404,198 @@ func TestApply(t *testing.T) {
+				k8sobjects.NamespaceObject("other-ns",
+					syncertest.ManagementEnabled,
+					core.Label("foo", "baz")),
+			},
+			declaredObjs: []client.Object{
+				newNamespaceObj(core.Name("deleted-ns"),
+					core.Label("foo", "bar"),
+					syncertest.ManagementEnabled,
+					syncertest.IgnoreMutationAnnotation)},
+			cachedIgnoredObjs: []client.Object{
+				&queue.Deleted{
+					Object: newNamespaceObj(
+						core.Name("deleted-ns"),
+						syncertest.ManagementEnabled,
+						syncertest.IgnoreMutationAnnotation)}},
+			expectedItemsInIgnoreCache: []client.Object{
+				&queue.Deleted{
+					Object: newNamespaceObj(
+						core.Name("deleted-ns"),
+						syncertest.ManagementEnabled,
+						syncertest.IgnoreMutationAnnotation)}},
+			expectedObjsToApply: object.UnstructuredSet{
+				asUnstructuredSanitizedObj(newNamespaceObj(core.Name("deleted-ns"),
+					syncertest.ManagementEnabled,
+					syncertest.IgnoreMutationAnnotation,
+					core.Label("foo", "bar"))),
+			},
+		},
+		{
+			name:       "mutation-ignored object doesn't currently exist on the cluster",
+			serverObjs: []client.Object{},
+			declaredObjs: []client.Object{
+				newNamespaceObj(core.Name("test-ns"),
+					syncertest.ManagementEnabled,
+					syncertest.IgnoreMutationAnnotation)},
+			cachedIgnoredObjs:          []client.Object{},
+			expectedItemsInIgnoreCache: nil,
+			expectedObjsToApply: object.UnstructuredSet{
+				asUnstructuredSanitizedObj(newNamespaceObj(core.Name("test-ns"),
+					syncertest.ManagementEnabled,
+					syncertest.IgnoreMutationAnnotation)),
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			rsObj := &unstructured.Unstructured{}
+			rsObj.SetGroupVersionKind(kinds.RepoSyncV1Beta1())
+			rsObj.SetNamespace(syncScope.SyncNamespace())
+			rsObj.SetName(rootSyncName)
+			tc.serverObjs = append(tc.serverObjs, rsObj)
+
+			syncScope := declared.Scope("test-namespace")
+			syncName := "rs"
+			fakeClient := testingfake.NewClient(t, core.Scheme, tc.serverObjs...)
+			fakeKptApplier := newFakeKptApplier([]event.Event{})
+			cs := &ClientSet{
+				KptApplier: fakeKptApplier,
+				Client:     fakeClient,
+				Mapper:     fakeClient.RESTMapper(),
+				// TODO: Add tests to cover status mode
+			}
+			var errs status.MultiError
+			eventHandler := func(event Event) {
+				if errEvent, ok := event.(ErrorEvent); ok {
+					if errs == nil {
+						errs = errEvent.Error
+					} else {
+						errs = status.Append(errs, errEvent.Error)
+					}
+				}
+			}
+
+			applier, err := NewNamespaceSupervisor(cs, syncScope, syncName, 5*time.Minute)
+			require.NoError(t, err)
+
+			resources := &declared.Resources{}
+			_, err = resources.UpdateDeclared(context.Background(), tc.declaredObjs, "")
+			require.NoError(t, err)
+			resources.UpdateIgnored(tc.cachedIgnoredObjs...)
+
+			applier.Apply(context.Background(), eventHandler, resources)
+
+			testutil.AssertEqual(t, tc.expectedObjsToApply, fakeKptApplier.objsToApply)
```

Comment:
Try to use `fakeClient.Check(t, tc.expectedServerObjs...)` so that we don't need to declare `expectedObjsToApply` as `object.UnstructuredSet`. Instead, it can be declared as `[]client.Object`, which is simpler.

Reply:
Right now, the objects are not actually applied to the fake server when the fake KptApplier is called. Should the fake KptApplier's Run fn be modified to actually apply the objects? Or should a function be added to the fake client that converts a given object to the scheme-preferred version so that it can be used in the tests?